### PR TITLE
Adding case folder details

### PR DESCRIPTION
**Pull request by tejans24**
connects #1998

### CODE REVIEW

File: app/models/issue.rb [hunk 1]
```diff
@@ -89,13 +87,11 @@ def load_from_vacols(hash)
...
       disposition = (VACOLS::Case::DISPOSITIONS[hash["issdc"]] || "other")
                     .parameterize.underscore.to_sym
-
       new(
-        description_label: (hash["isscode_label"]).to_s,
         levels: parse_levels_from_vacols(hash),
         vacols_sequence_id: hash["issseq"],
-        program: PROGRAMS[hash["issprog"]],
-        type: TYPES[hash["isscode"]],
+        program: { name: PROGRAMS[hash["issprog"]], label: hash["issprog_label"] },
```
Comment (mdbenjam) [inline app/models/issue.rb]: I think this might be used in `HearingWorksheet.jsx` Let's make sure we don't break them.

File: app/models/vacols/case.rb [hunk 2]
```diff
@@ -234,5 +234,39 @@ def update_vacols_location!(location)
     end
   end
 
+  def self.aod(vacols_id)
+    conn = connection
+    vacols_id = conn.quote(vacols_id)
+
+    conn.transaction do
+      query = <<-SQL
+        SELECT (case when (nvl(AOD_DIARIES.CNT, 0) + nvl(AOD_HEARINGS.CNT, 0)) > 0 then 1 else 0 end) AOD
```
Comment (mdbenjam) [inline app/models/vacols/case.rb]: Did @cmgiven write this. I have no idea how to read it haha. So as long as he's good with it, it's good with me.

File: client/app/reader/ClaimsFolderDetails.jsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    className: 'claim-detail-aod' },
+  none: { text: '',
+    className: '' }
+};
+
+class ClaimsFolderDetails extends React.PureComponent {
+
+  getClaimTypeDetailInfo() {
+    const { appeal } = this.props;
+    let obj = TYPE_INFO.none;
```
Comment (mdbenjam) [inline client/app/reader/ClaimsFolderDetails.jsx]: Can we use a more descriptive variable here?

File: app/models/vacols/case.rb (hunk 2 above)
Comment (cmgiven) [inline app/models/vacols/case.rb]: Yes, I write the code that makes ya'll gasp in horror.

I guess I do have one concern here on performance. This looks like it will be called once for each case, each time the welcome gate is viewed. This is a query with two joins. Is there any way to cache this value? Or do some ActiveRecord magic so that it doesn't get run for each individual case? (We could add it into the standard joins, but then we're grabbing it always; I'm not sure if there's a middle ground.) Once a case is AOD according to this query, it should remain AOD for perpetuity.

Comment (tejans24) [inline app/models/vacols/case.rb]: If the performance is a concern, the alternative is to cache it or store it in the Caseflow db. I can also see if there are any other alternatives with ActiveRecord.

File: client/app/components/Accordion.jsx [hunk 4]
```diff
@@ -18,11 +18,12 @@ export default class Accordion extends React.PureComponent {
...
-      style
+      style,
+      id
     } = this.props;
-
-    const accordionHeaders = children.map((child) => {
-      return <Panel header={child.props.title} headerClass="usa-accordion-button" key={child.props.title}>
+    const arrayOfChildren = React.Children.toArray(children);
+    const accordionHeaders = arrayOfChildren.map((child) => {
+      return <Panel id={id} header={child.props.title} headerClass="usa-accordion-button" key={child.props.title}>
```
Comment (tejans24) [inline client/app/components/Accordion.jsx]: oops, good catch. I will change that.

File: client/app/components/Accordion.jsx [hunk 5]
```diff
@@ -18,11 +18,12 @@ export default class Accordion extends React.PureComponent {
...
     const {
       children,
-      style
+      style,
+      id
     } = this.props;
-
-    const accordionHeaders = children.map((child) => {
-      return <Panel header={child.props.title} headerClass="usa-accordion-button" key={child.props.title}>
+    const arrayOfChildren = React.Children.toArray(children);
```
Comment (tejans24) [inline client/app/components/Accordion.jsx]: children is not guaranteed to be an array here if there is only one object. I think there is a issue in React to change that but for now it needs to be converted to an array.

File: app/models/vacols/case.rb (hunk 2 above)
Comment (mdbenjam) [inline app/models/vacols/case.rb]: I vote we cache it, but also think we should do it in another ticket. As long as the performance concerns aren't dire (which I don't think they are).

File: client/app/reader/ClaimsFolderDetails.jsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+              </ol>
+          </div>
+        </AccordionHeader>
+      </Accordion>
+    </div>;
+  }
+}
+
+const mapDispatchToProps = (dispatch) => ({
+  handleToggleCommentOpened(docId) {
```
Comment (mdbenjam) [inline client/app/reader/ClaimsFolderDetails.jsx]: Where is this being called?

File: client/app/reader/DecisionReviewer.jsx [hunk 7]
```diff
@@ -181,7 +180,8 @@ const mapStateToProps = (state) => {
   return {
     documentFilters: state.ui.pdfList.filters,
     storeDocuments: state.documents,
-    pdf: state.ui.pdf
+    pdf: state.ui.pdf,
+    assignments: state.assignments
```
Comment (mdbenjam) [inline client/app/reader/DecisionReviewer.jsx]: Where are we using this?

File: client/app/reader/actions.js [hunk 8]
```diff
@@ -422,6 +423,28 @@ export const removeTag = (doc, tagId) => (
...
 );
 
+
+export const onReceiveAppealDetails = (appeal) => ({
+  type: Constants.RECEIVE_APPEAL_DETAILS,
+  payload: { appeal }
+});
+
+export const onAppealDetailsLoadingFail = () => ({
+  type: Constants.REQUEST_INITIAL_DATA_FAILURE
```
Comment (mdbenjam) [inline client/app/reader/actions.js]: Do we do anything in the failure case?

File: client/app/reader/ClaimsFolderDetails.jsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    return <span className={obj.className}>{obj.text}</span>;
+  }
+
+  render() {
+    const { appeal } = this.props;
+
+    if (_.isEmpty(appeal)) {
+      return <p className="loading-text">Loading...</p>;
```
Comment (mdbenjam) [inline client/app/reader/ClaimsFolderDetails.jsx]: I think this makes more sense in the accordion. That way people don't see it while it's loading unless they go looking for it.

File: lib/fakes/appeal_repository.rb [hunk 10]
```diff
@@ -358,7 +358,51 @@ def self.seed_reader_data!
...
         veteran_first_name: "Joe",
-        veteran_last_name: "Smith"
+        veteran_last_name: "Smith",
+        type: "Court Remand",
+        date_assigned: "2013-05-17 00:00:00 UTC".to_datetime,
+        date_received: "2013-05-31 00:00:00 UTC".to_datetime,
+        signed_date: nil,
+        docket_number: "13 11-265",
+        regional_office_key: "RO13",
+        issues: [
```
Comment (mdbenjam) [inline lib/fakes/appeal_repository.rb]: Is there a way to use the issue generator here?

File: lib/fakes/appeal_repository.rb [hunk 11]
```diff
@@ -358,7 +358,51 @@ def self.seed_reader_data!
       vacols_record: {
         template: :ready_to_certify,
         veteran_first_name: "Joe",
-        veteran_last_name: "Smith"
+        veteran_last_name: "Smith",
```
Comment (mdbenjam) [inline lib/fakes/appeal_repository.rb]: Is there anyway to define some of this information once and use it multiple times?

File: spec/models/appeal_spec.rb [hunk 12]
```diff
@@ -924,6 +924,29 @@
       end
     end
 
+    context "Appel includes addtional required fields" do
```
Comment (mdbenjam) [inline spec/models/appeal_spec.rb]: I would combine this with the above context. "additional required fields" sounds a little janky.

File: spec/models/appeal_spec.rb [hunk 13]
```diff
@@ -924,6 +924,29 @@
...
+          docket_number: "13 11-265",
+          regional_office_key: "RO13",
+          type: "Court Remand",
+          vacols_record: {
+            soc_date: 4.days.ago
+          }
+        )
+      end
+
+      it "includes aod, cavc, regional_office and docket_number" do
```
Comment (mdbenjam) [inline spec/models/appeal_spec.rb]: do we have other tests for the new methods like `aod`, `cavc`, `regional office` etc.

File: client/app/components/Accordion.jsx (hunk 5 above)
Comment (mdbenjam) [inline client/app/components/Accordion.jsx]: Can you give it a shot. I agree that children.map is wrong, but I think React.children.map is meant for this case.

Comment (tejans24) [inline client/app/components/Accordion.jsx]: I misread your comment earlier, you are right React.Children.map works in this case.

File: app/models/vacols/case.rb (hunk 2 above)
Comment (tejans24) [inline app/models/vacols/case.rb]: I created an issue for this: https://github.com/department-of-veterans-affairs/caseflow/issues/2674

File: client/app/reader/ClaimsFolderDetails.jsx (hunk 9 above)
Comment (tejans24) [inline client/app/reader/ClaimsFolderDetails.jsx]: This might need a bit more design input since the Header above the accordion requires some Veteran information fetched from the API call.

File: client/app/reader/DecisionReviewer.jsx (hunk 7 above)
Comment (tejans24) [inline client/app/reader/DecisionReviewer.jsx]: I changed how we fetched the appeal information and forgot to remove this....grr.

File: client/app/reader/actions.js (hunk 8 above)
Comment (tejans24) [inline client/app/reader/actions.js]: yea, I'm getting a failure state UI from Gina tomorrow. Most of the code is ready to implement this.

File: client/app/components/Accordion.jsx [hunk 14]
```diff
@@ -21,14 +21,16 @@ export default class Accordion extends React.PureComponent {
       children,
       ...passthroughProps
     } = this.props;
+    const accordionSections = React.Children.map(children, (child) => {
+      let headerClass = "usa-accordion-button";
```
Comment (NickHeiner) [inline client/app/components/Accordion.jsx]: Please default to `const`.

File: client/app/components/Accordion.jsx [hunk 15]
```diff
@@ -21,14 +21,16 @@ export default class Accordion extends React.PureComponent {
...
 
-    const accordionSections = children.map((child) => {
-      return <Panel header={child.props.title} headerClass="usa-accordion-button"
-        key={child.props.title} id={child.props.id}>
-          <div className="usa-accordion-content">
-            {child.props.children}
-          </div>
-        </Panel>;
+      return <Panel id={child.props.id} showArrow={false} 
+        headerClass={child.props.loading ? `${headerClass} loading` : headerClass}
```
Comment (NickHeiner) [inline client/app/components/Accordion.jsx]: Instead of this ternary, use `classnames`.

```js
classnames(headerClass, {
    loading: child.props.loading
})
```

File: client/app/reader/ClaimsFolderDetails.jsx [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+              <span>{appeal.docket_number}</span>
+            </div>
+            <div className="usa-width-one-fourth">
+              <b>Regional Office</b><br />
+              <span>{`${appeal.regional_office.key} - ${appeal.regional_office.city}`}</span>
+            </div>
+            <div className="usa-width-one-whole claims-folder-issues">
+                <b>Issues</b><br />
+                <ol>
+                  {appeal.issues && appeal.issues.map((issue, index) =>
```
Comment (NickHeiner) [inline client/app/reader/ClaimsFolderDetails.jsx]: If we do `_.map`, then we don't need the `appeal.issues &&` at the beginning.

File: client/app/reader/ClaimsFolderDetails.jsx [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+                    <li key={index}><span>
```
Comment (NickHeiner) [inline client/app/reader/ClaimsFolderDetails.jsx]: Is there anything else we can key off of? There's no `id`-like field in `issue`?

The problem with doing this is that if we ever insert or delete issues, we'll re-render every issue after that one in the table. This is not a problem given how trivial the size of `issues` is likely to be, but if there's already a unique `id` field on `issue`, we should just use that instead.

File: client/app/reader/PdfListView.jsx [hunk 18]
```diff
@@ -35,13 +47,22 @@ export class PdfListView extends React.Component {
...
 
-const mapStateToProps = (state) => ({
-  documents: getFilteredDocuments(state),
-  ..._.pick(state.ui, 'docFilterCriteria')
-});
+const mapStateToProps = (state, props) => {
+  return { documents: getFilteredDocuments(state),
+    ..._.pick(state.ui, 'docFilterCriteria'),
+    appeal: _.find(state.assignments, { vacols_id: props.match.params.vacolsId }) ||
+    state.loadedAppeal
```
Comment (NickHeiner) [inline client/app/reader/PdfListView.jsx]: Can you indent this another level? As it is, it looks like it's another property in the top-level object.

```js
appeal: _.find(state.assignments, { vacols_id: props.match.params.vacolsId }) || 
  state.loadedAppeal
```

File: client/app/reader/actions.js [hunk 19]
```diff
@@ -425,6 +425,29 @@ export const removeTag = (doc, tagId) => (
...
+  payload: { value }
+});
+
+export const fetchAppealDetails = (vacolsId) => (
+  (dispatch) => {
+    ApiUtil.get(`/reader/appeal/${vacolsId}?json`).then((response) => {
+      const returnedObject = JSON.parse(response.text);
+
+      dispatch(onReceiveAppealDetails(returnedObject.appeal));
+    }, () => {
```
Comment (NickHeiner) [inline client/app/reader/actions.js]: Please do not use `{` when the function only has one expression. Please use the shorthand instead.

File: client/app/reader/actions.js [hunk 20]
```diff
@@ -425,6 +425,29 @@ export const removeTag = (doc, tagId) => (
...
 
+
+export const onReceiveAppealDetails = (appeal) => ({
+  type: Constants.RECEIVE_APPEAL_DETAILS,
+  payload: { appeal }
+});
+
+export const onAppealDetailsLoadingFail = (value = true) => ({
+  type: Constants.RECEIVE_APPEAL_DETAILS_FAILURE,
+  payload: { value }
```
Comment (NickHeiner) [inline client/app/reader/actions.js]: `value` is not a descriptive name. Can you pick a name that communicates more what the payload is?

File: client/app/reader/reducer.js [hunk 21]
```diff
@@ -292,6 +294,22 @@ export const reducer = (state = initialState, action = {}) => {
...
+      {
+        loadedAppeal: {
+          $set: action.payload.appeal
+        }
+      }
+    );
+  case Constants.RECEIVE_APPEAL_DETAILS_FAILURE:
+    return update(
+      {
+        loadedAppealFail: {
```
Comment (NickHeiner) [inline client/app/reader/reducer.js]: If this is a boolean that tracks whether the appeal failed to load, how about `didLoadedAppealFail`? I like `did` or `is` at the beginning of a boolean var to make it clearer that the variable is the answer to a question.

File: client/app/reader/ClaimsFolderDetails.jsx (hunk 17 above)
Comment (cmgiven) [inline client/app/reader/ClaimsFolderDetails.jsx]: FYI: Issues are uniqued on the combination of `ISSUES.ISSKEY` (the VACOLS appeal ID) and `ISSUES.ISSSEQ`.

File: lib/fakes/case_assignment_repository.rb [hunk 22]
```diff
@@ -6,32 +6,137 @@ class << self
...
+              },
+              type: { name: :service_connection, label: "Service Connection" }, category: :knee, levels: [
+                "Lumbosacral",
+                "All Others",
+                "Thigh, limitation of flexion of"
+              ] },
+            { disposition: :remanded, program:
+              {
+                name: :compensation,
+                label: "Compenstation"
```
Comment (NickHeiner) [inline lib/fakes/case_assignment_repository.rb]: typo: "Compenstation"

File: spec/feature/reader_spec.rb [hunk 23]
```diff
@@ -749,6 +749,35 @@ def get_aria_labels(elems)
       expect(find("#other", visible: false).checked?).to be false
     end
 
+    scenario "Claim Folder Details" do
+      visit "/reader/appeal/#{appeal.vacols_id}/documents"
+
+      vetern_folder_title = "#{appeal.veteran_full_name}'s Claims Folder"
```
Comment (NickHeiner) [inline spec/feature/reader_spec.rb]: typo: "vetern"

File: spec/models/appeal_spec.rb [hunk 24]
```diff
@@ -905,13 +905,18 @@
   end
 
   context ".to_hash" do
-    context "when issues parameter is nil" do
+    context "when issues parameter is nil and contains addtional attributes" do
```
Comment (NickHeiner) [inline spec/models/appeal_spec.rb]: typo: "addtional"

File: lib/fakes/appeal_repository.rb [hunk 25]
```diff
@@ -404,9 +416,20 @@ def self.seed_reader_data!
...
+        docket_number: "13 11-265",
+        regional_office_key: "RO13",
       },
+      issues: [Generators::Issue.build(vacols_id: "reader_id1")],
       documents: redacted_reader_documents
     )
   end
+
+  def self.aod(_vacols_id)
+    1
```
Comment (mdbenjam) [inline lib/fakes/appeal_repository.rb]: Maybe enable this to be overriden by a class variable so that our test can be less dependent on the repository.

File: spec/models/appeal_spec.rb [hunk 26]
```diff
@@ -101,6 +101,26 @@
...
+    it { is_expected.to be_truthy}
+  end
+
+  context "#cavc" do
+    subject { appeal }
+
+    it { expect(subject.cavc).to be_falsey }
+
+    context "when appeal type is court remand" do
+      let(:type) { "Court Remand" }
```
Comment (mdbenjam) [inline spec/models/appeal_spec.rb]: you're not using the type variable you just let.

File: client/app/reader/reducer.js [hunk 27]
```diff
@@ -292,6 +294,22 @@ export const reducer = (state = initialState, action = {}) => {
           $set: action.payload.assignments
         }
       });
+  case Constants.RECEIVE_APPEAL_DETAILS:
+    return update(state,
+      {
+        loadedAppeal: {
+          $set: action.payload.appeal
```
Comment (mdbenjam) [inline client/app/reader/reducer.js]: Why not just set this in the assignments array?

File: app/models/issue.rb [hunk 28]
```diff
@@ -89,13 +87,11 @@ def load_from_vacols(hash)
...
       disposition = (VACOLS::Case::DISPOSITIONS[hash["issdc"]] || "other")
                     .parameterize.underscore.to_sym
-
       new(
-        description_label: (hash["isscode_label"]).to_s,
         levels: parse_levels_from_vacols(hash),
         vacols_sequence_id: hash["issseq"],
         program: PROGRAMS[hash["issprog"]],
-        type: TYPES[hash["isscode"]],
+        type: { name: TYPES[hash["isscode"]], label: hash["isscode_label"] },
```
Comment (mdbenjam) [inline app/models/issue.rb]: Does this impact other apps?

Comment (tejans24) [inline app/models/issue.rb]: I verified that it's currently not being used anywhere else

File: client/app/components/Accordion.jsx [hunk 29]
```diff
@@ -19,12 +20,16 @@ export default class Accordion extends React.PureComponent {
...
     } = this.props;
-    const arrayOfChildren = React.Children.toArray(children);
-    const accordionHeaders = arrayOfChildren.map((child) => {
-      return <Panel id={id} header={child.props.title} headerClass="usa-accordion-button" key={child.props.title}>
+    const accordionSections = React.Children.map(children, (child) => {
+      const headerClass = 'usa-accordion-button';
+
+      return <Panel id={child.props.id} showArrow={false} 
+        headerClass={classnames(headerClass, {
+          loading: child.props.loading
```
Comment (tejans24) [inline client/app/components/Accordion.jsx]: yea, that makes sense but I'll quickly verify with @gnakm to see if that works.

File: client/app/reader/reducer.js (hunk 27 above)
Comment (tejans24) [inline client/app/reader/reducer.js]: Because the loaded appeal could be outside of what's assigned to an attorney. It might be a bit confusing in the Welcome gate case assignments. I guess we can keep track of wether the appeal is assigned or not with a boolean but I felt like that adds a bit more complexity.

File: lib/fakes/appeal_repository.rb (hunk 25 above)
Comment (tejans24) [inline lib/fakes/appeal_repository.rb]: when @repository.aod is called, this method is expected in the Appeal model.

File: app/assets/stylesheets/_reviewer.scss [hunk 30]
```diff
@@ -56,9 +56,15 @@ $bookmark-yellow: #efdf1a;
...
 }
 
+
+.rc-collapse > .rc-collapse-item-disabled > .rc-collapse-header {
+  color: black;
+}
+
 [id=claim-folder-details-accordion] {
-  .loading {
+  .disabled {
```
Comment (mdbenjam) [inline app/assets/stylesheets/_reviewer.scss]: It's unclear to me that this is needed. I think the disabled quality is enough. I don't know that we need to remove the + sign.

File: app/assets/stylesheets/_reviewer.scss [hunk 31]
```diff
@@ -56,9 +56,15 @@ $bookmark-yellow: #efdf1a;
   }
 }
 
+
+.rc-collapse > .rc-collapse-item-disabled > .rc-collapse-header {
```
Comment (mdbenjam) [inline app/assets/stylesheets/_reviewer.scss]: This seems like a better candidate for the accordion specific css file.

File: client/app/reader/reducer.js (hunk 27 above)
Comment (mdbenjam) [inline client/app/reader/reducer.js]: My concern is that we have to send off a new request even if it's a page we've been to before. But you're right it keeps things simpler... so let's leave it for now and optimize later if it is an actual problem.

File: lib/fakes/appeal_repository.rb (hunk 25 above)
Comment (mdbenjam) [inline lib/fakes/appeal_repository.rb]: That's fine, I'm just suggesting something like an attr_accessor for a variable to override this so we can test non aod cases. It just makes our appeal tests more explicit since we can setup true or false.

File: app/controllers/reader/appeal_controller.rb [hunk 32]
```diff
@@ -11,4 +11,19 @@ def index
       end
     end
   end
+
+  def show
+    vacols_id = params[:id]
+
+    respond_to do |format|
+      format.json do
+        MetricsService.record "Get appeal information for #{vacols_id}" do
```
Comment (joofsh) [inline app/controllers/reader/appeal_controller.rb]: can we add the `service` and `name` parameters to the `record()` call here please

File: app/models/appeal.rb [hunk 33]
```diff
@@ -192,6 +195,14 @@ def station_key
     result && result.first
   end
 
+  def cavc
+    type == VACOLS::Case::TYPES[CAVC_TYPE]
+  end
+
+  def aod
+    self.class.repository.aod(vacols_id) == 1
+  end
```
Comment (joofsh) [inline app/models/appeal.rb]: for this and the `cavc` method above. We should try and keep this sort of VACOLS logic outside of our Caseflow appeal. Could we have `self.class.repository.aod(vacols_id)` itself do the `== 1` comparison so it returns a pure `true/false`

For cavc, can we instead move this to our base appeal <=> vacols mapper here: https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/services/appeal_repository.rb#L75-L108 , and then `cavc` would be another vacols_attr_accessor

File: app/models/vacols/case.rb (hunk 2 above)
Comment (joofsh) [inline app/models/vacols/case.rb]: It's probably worth caching this in our caseflow DB too, but as an intermediate step in this PR, can we memoize it within the object? ` @aod ||=  <method call>`. That way at most the query is called only 1 time per request cycle

File: client/app/reader/constants.js [hunk 34]
```diff
@@ -68,7 +68,7 @@ export const SET_PAGE_COORD_BOUNDS = 'SET_PAGE_COORD_BOUNDS';
 
 export const SET_OPENED_ACCORDION_SECTIONS = 'SET_OPENED_ACCORDION_SECTIONS';
 
-export const COMMENT_ACCORDION_KEY = 'Comments';
+export const COMMENT_ACCORDION_KEY = 'comments-header';
```
Comment (NickHeiner) [inline client/app/reader/constants.js]: Why did this need to change?

File: app/services/appeal_repository.rb [hunk 35]
```diff
@@ -104,7 +106,9 @@ def self.set_vacols_values(appeal:, case_record:)
       prior_decision_date: normalize_vacols_date(case_record.bfdpdcn),
       status: VACOLS::Case::STATUS[case_record.bfmpro],
       outcoding_date: normalize_vacols_date(folder_record.tioctime),
-      private_attorney_or_agent: case_record.bfso == "T"
+      private_attorney_or_agent: case_record.bfso == "T",
+      docket_number: folder_record.ticknum,
+      cavc: VACOLS::Case::TYPES[case_record.bfac] == VACOLS::Case::TYPES[CAVC_TYPE]
```
Comment (tejans24) [inline app/services/appeal_repository.rb]: I'm adding a test for this

File: client/app/components/Accordion.jsx [hunk 36]
```diff
@@ -23,16 +24,17 @@ export default class Accordion extends React.PureComponent {
       ...passthroughProps
     } = this.props;
 
-    console.log(passthroughProps)
-    const accordionSections = React.Children.map(children, (child) => {
+    // converting children to array and mapping over them
+    const accordionSections = _.map(React.Children.toArray(children), (child) => {
```
Comment (mdbenjam) [inline client/app/components/Accordion.jsx]: Why this change?

File: client/app/components/AccordionSection.jsx [hunk 37]
```diff
@@ -9,5 +9,8 @@ export default class AccordionSection extends React.PureComponent {
 
 AccordionSection.propTypes = {
   children: PropTypes.node,
-  title: PropTypes.string
+  title: PropTypes.node,
```
Comment (mdbenjam) [inline client/app/components/AccordionSection.jsx]: Why did you change this to a node?

File: client/app/components/AccordionSection.jsx [hunk 38]
```diff
@@ -9,5 +9,8 @@ export default class AccordionSection extends React.PureComponent {
...
+  disabled: PropTypes.bool,
+  id: PropTypes.string,
+  sectionKey: PropTypes.string
```
Comment (mdbenjam) [inline client/app/components/AccordionSection.jsx]: Can we no longer use the title for a key because it's a node?

File: client/app/components/AccordionSection.jsx (hunk 37 above)
Comment (tejans24) [inline client/app/components/AccordionSection.jsx]: Merge issue from another PR: i'll change it to a string again, it will be changed to a node in another PR.

File: client/app/components/Accordion.jsx (hunk 36 above)
Comment (tejans24) [inline client/app/components/Accordion.jsx]: because React.Children.map suffixes the keys by default with /index. A "document-information" key passed in becomes "document-information/0", "categories" -> "categories/1". 

This requires a change in openedAccordionSections key names to match those key names. I thought this is a better alternative.

File: client/app/components/AccordionSection.jsx [hunk 39]
```diff
@@ -9,5 +9,8 @@ export default class AccordionSection extends React.PureComponent {
 
 AccordionSection.propTypes = {
   children: PropTypes.node,
-  title: PropTypes.string
+  title: PropTypes.string,
+  disabled: PropTypes.bool,
+  id: PropTypes.string,
+  sectionKey: PropTypes.string
```
Comment (mdbenjam) [inline client/app/components/AccordionSection.jsx]: We should just be using the title.

File: client/app/components/Accordion.jsx (hunk 36 above)
Comment (mdbenjam) [inline client/app/components/Accordion.jsx]: Is this changes still necessary even after removing the keys?

**Comment by tejans24** [comment]
Yes, that change is still required. It has more to do with React.Children.map function that's suffixing index to the key.

File: lib/fakes/appeal_repository.rb (hunk 11 above)
Comment (tejans24) [inline lib/fakes/appeal_repository.rb]: yes it can but I think it needs to be another ticket to consolidate the appeal fakes being used in different places in Reader and Caseflow.

Comment (mdbenjam) [inline lib/fakes/appeal_repository.rb]: Okay let's make a tech improvement ticket for this.

File: lib/fakes/appeal_repository.rb (hunk 25 above)
Comment (tejans24) [inline lib/fakes/appeal_repository.rb]: https://github.com/department-of-veterans-affairs/caseflow/issues/2755 - I created an issue for this

File: lib/fakes/appeal_repository.rb (hunk 11 above)
Comment (tejans24) [inline lib/fakes/appeal_repository.rb]: https://github.com/department-of-veterans-affairs/caseflow/issues/2756
